rfcs/0000-rfc-process.md

```diff
@@ -0,0 +1,160 @@
+- Start Date: 2019-03-22
+- RFC PR: (leave this blank)
+- cargo/crates.io issue: (leave this blank)
+
+# Summary
+
+This describes the RFC process for the crates.io and Cargo teams, separate from
+the general RFC process. While it is expected that the majority of RFCs opened
+here would come from members of the crates.io or cargo teams, new RFCs and
+discussion from others is both welcomed and encouraged.
+
+# Motivation
+
+The goal of creating a separate RFC repo and process for these teams is to
+increase communication, while decreasing noise in the main RFC repo. A large
+number of changes would benefit from going through the RFC process, but either
+only affect the two teams, or are irrelevant to the majority of Rust's users.
+
+By providing our own RFC repo, we are able to give a better process to propose
+changes which affect both teams, and give users who are interested in lower
+level changes a single repo to watch that doesn't include noise from bug fixes,
+etc.
+
+Finally, this allows us to propose changes which we wish to give users a chance
+to chime in on, without requiring them to attend a team meeting. This also
+reduces reliance on synchronous meetings for members of the teams, which are
+becoming increasingly distributed.
+
+# Impact on both teams
+
+Many things which are currently only discussed in issues/PRs or team meetings
+would be expected to go through an RFC in this repo. RFCs opened against the
+main repo assigned to either team may be closed if they are better suited to be
+opened here.
+
+Many RFCs should still be opened on the main repo. When it's unclear where it
+belongs, it's left to the discretion of the teams to decide which location is
+more appropriate.
```

Comment:
In some cases do you think it would be helpful to open an RFC on the main repo and a sub-RFC here?  For example, the main RFC could cover user facing command line interface and `Cargo.toml` details, and link to an RFC here for details on API and index changes.

Side note: I think as part of this process we will want to make clear when we are committing to stability for general use (other tools talking to crates.io, and cargo to alternative registries), and when details are an internal implementation between Cargo and crates.io.  Hopefully better namespacing under `/api/` for any new endpoints will go a long way to documenting this.

Reply:
I think for now it's better to keep everything in a single RFC per feature, but this is certainly something we can experiment with. I don't think we need to outline an explicit process for that here though